src/main/ruby/core/truffle/kernel_operations.rb

```diff
@@ -107,5 +107,41 @@ def self.define_read_only_global(name, getter)
       -> { global_variable_get(:$stderr) },
       -> v { raise TypeError, "$stderr must have a write method #{v.class} given." unless v.respond_to?(:write)
              global_variable_set(:$stderr, v) })
+   
+     def self.internal_raise(exc, msg, ctx=nil, internal)
```

Comment:
Having a lone optional parameter in the middle of the parameter list looks odd to me. But, maybe that's just a stylistic thing.

Reply:
I would drop the default value, since there are only 2 callers and it's internal API anyway so it doesn't have to be as pretty.